src/metric-engine/src/engine/create.rs

```diff
@@ -600,17 +597,17 @@ fn parse_physical_region_id(request: &RegionCreateRequest) -> Result<RegionId> {
 
 /// Creates the region options for metadata region in metric engine.
 pub(crate) fn region_options_for_metadata_region(
-    mut original: HashMap<String, String>,
+    original: &HashMap<String, String>,
 ) -> HashMap<String, String> {
     // TODO(ruihang, weny): add whitelist for metric engine options.
-    original.remove(APPEND_MODE_KEY);
-    // Don't allow to set primary key encoding for metadata region.
-    original.remove(MEMTABLE_PARTITION_TREE_PRIMARY_KEY_ENCODING);
-    // Don't allow to set memtable type for metadata region.
-    original.remove(MEMTABLE_TYPE);
-    original.insert(TTL_KEY.to_string(), FOREVER.to_string());
-    original.remove(SKIP_WAL_KEY);
-    original
+    let mut metadata_region_options = HashMap::new();
+    metadata_region_options.insert(TTL_KEY.to_string(), FOREVER.to_string());
+
+    if let Some(wal_options) = original.get(WAL_OPTIONS_KEY) {
+        metadata_region_options.insert(WAL_OPTIONS_KEY.to_string(), wal_options.to_string());
+    }
```

Comment:
Should we reuse the compaction options for the data region? Maybe the metadata region should have different compaction options because it mainly stores KV pairs.

Reply:
meatadata region should not inherit the compaction parameters. Maybe we can manually specify a metadata region specific compaction parameter set in the future.